src/System.Runtime.Serialization.Formatters/tests/TypeSerializableValue.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+
+namespace System.Runtime.Serialization.Formatters.Tests
+{
+    public readonly struct TypeSerializableValue
+    {
+        public readonly string Base64Blob;
+        public readonly TargetFrameworkMoniker Platform;
+
+        public TypeSerializableValue(string base64Blob, TargetFrameworkMoniker platform)
+        {
+            Base64Blob = base64Blob;
+            Platform = platform;
+        }
+
+        public static int GetPlatformIndex(TypeSerializableValue[] blobs)
+        {
+            var blobList = blobs.ToList();
+            int index;
+            // .NET Framework
+            if (PlatformDetection.IsFullFramework)
+            {
+                // Check if a specialized blob for >=netfx471 is present and return if found.
+                if (PlatformDetection.IsNetfx471OrNewer)
+                {
+                    index = blobList.FindIndex(b => b.Platform == TargetFrameworkMoniker.netfx471);
+
+                    if (index >= 0)
+                        return index;
+                }
+                    
+                // If no newer blob for >=netfx471 is present use existing one. 
+                // If no netfx blob is present then -1 will be returned.
+                return blobList.FindIndex(b => b.Platform == TargetFrameworkMoniker.netfx);
+            }
+
+            // .NET Core
+            if (PlatformDetection.IsNetCore)
+            {
+                // Check if a specialized blob for >=netcoreapp2.1 is present and return if found.
+                index = blobList.FindIndex(b => b.Platform == TargetFrameworkMoniker.netcoreapp21);
+
+                if (index >= 0)
+                    return index;
+            }
+
+            // If no newer blob for >=netcoreapp2.1 is present use existing one. 
+            // If no netfx blob is present then -1 will be returned.
+            return blobList.FindIndex((b => b.Platform == TargetFrameworkMoniker.netcoreapp));
+        }
+    }
+
+    public enum TargetFrameworkMoniker
+    {
+        netfx,
+        netfx471,
```

Comment:
For most of the repo netfx is an alias for the latest which in this case is net471 and netcoreapp is an alias for netcorepp2.1 which is the latest. Is there a reason you need the versions here? I would suggest either going all versioned or not.

Reply:
The version tells us when a particular serialization blob was changed.
We can change  it to 
```
    public enum TargetFrameworkMoniker
    {
        netfx470orOlder,
        netfx471orNewer,
        netcoreapp20orOlder,
        netcoreapp21orNewer,
    }
```
?

Reply:
I'm not sure why some are Older and some are Newer. Should it be a min version of when the blob changed?

At any rate I think having more defined names like your suggestion will be clearer.

Reply:
yes it should be the min version when the blob changed. The blobs only changed in netfx471 and netcoreapp2.1 in terms of serialization

Reply:
What about:

```
netfx470
netfx471
netcoreapp20 (not older, as we didn't support serialization before)
netcoreapp21
…
```

Reply:
hit enter too early. updated the previous comment.